### PR TITLE
feat!: Forbid BSX XCM transfer

### LINKED ISSUE #448: Forbid Basilisk (native asset) transfer out

### PR DESCRIPTION
**Pull request by apopiak**
This PR deactivates XCM support for BSX by adding a filter to pallet_xcm as well as removing the mapping between the `CORE_ASSET_ID` and its corresponding `MultiLocation`.

### Notes
+ the testing runtime keeps BSX reserve transfer support
+ adjusted the runtime integration tests a bunch
+ added `std` feature to `common-runtime` crate so it compiles the test correctly

## Related Issue

https://app.clickup.com/t/2mrhgw8
Closes #448 

## Motivation and Context

For the release we want to forbid moving BSX out of the chain.

## How Has This Been Tested?

Added a runtime integration test.

## Checklist:

- [x] I have updated the documentation if necessary.
- [x] I have added tests to cover my changes, regression test if fixing an issue.
- [x] This is a breaking change.

### CODE REVIEW

**Comment by lumir-mrkva** [comment]
Task linked: [CU-2mrhgw8 Forbid BSX (native asset) from being transfered out via XCM](https://app.clickup.com/t/2mrhgw8)

File: runtime/basilisk/src/lib.rs [hunk 1]
```diff
@@ -66,7 +66,7 @@ use hydradx_traits::AssetPairAccountIdFor;
 use pallet_transaction_payment::TargetedFeeAdjustment;
 pub use sp_consensus_aura::sr25519::AuthorityId as AuraId;
 
-mod xcm;
+pub mod xcm;
```
Comment (apopiak) [inline runtime/basilisk/src/lib.rs:69]: made this public so we can access public config values like `NativeLocation`

File: runtime/basilisk/src/xcm.rs [hunk 2]
```diff
@@ -211,13 +232,13 @@ impl Convert<AssetId, Option<MultiLocation>> for CurrencyIdConvert {
...
-				// Handling native asset for this parachain
-				Some(CORE_ASSET_ID)
-			}
+			// MultiLocation {
+			// 	parents,
+			// 	interior: X2(Parachain(id), GeneralIndex(index)),
+			// } if parents == 1 && ParaId::from(id) == ParachainInfo::get() && (index as u32) == CORE_ASSET_ID => {
+			// 	// Handling native asset for this parachain
+			// 	Some(CORE_ASSET_ID)
+			// }
```
Comment (apopiak) [inline runtime/basilisk/src/xcm.rs:241]:
```suggestion
```

File: integration-tests/Cargo.toml [hunk 3]
```diff
@@ -108,6 +108,7 @@ kusama-runtime = { git = "https://github.com/paritytech/polkadot", branch = "rel
 
 [dev-dependencies]
 xcm-emulator = { git = "https://github.com/zqhxuyuan/xcm-simulator", branch = "v0.9.16" }
+env_logger = "*"
```
Comment (apopiak) [inline integration-tests/Cargo.toml:111]:
```suggestion
```

File: integration-tests/src/kusama_test_net.rs [hunk 4]
```diff
@@ -185,7 +207,7 @@ pub fn basilisk_ext() -> sp_io::TestExternalities {
 	.unwrap();
 
 	pallet_asset_registry::GenesisConfig::<Runtime> {
-		asset_names: vec![(b"KSM".to_vec(), 1_000_000u128)],
+		asset_names: vec![(b"KSM".to_vec(), 1_000_000u128), (b"TRF".to_vec(), 1_000_000u128)],
```
Comment (apopiak) [inline integration-tests/src/kusama_test_net.rs:210]:
```suggestion
		asset_names: vec![(b"KSM".to_vec(), 1_000_000u128), (b"SNEK".to_vec(), 1_000_000u128)],
```

File: runtime/basilisk/Cargo.toml [hunk 5]
```diff
@@ -20,6 +20,7 @@ serde = { features = ["derive"], optional = true, version = "1.0.136" }
 codec = { package = "parity-scale-codec", version = "2.3.1", default-features = false, features = ["derive"] }
 scale-info = { version = "1.0", default-features = false, features = ["derive"] }
 smallvec = "1.6.1"
+log = { version = "0.4.14", default-features = false }
```
Comment (apopiak) [inline runtime/basilisk/Cargo.toml:23]:
```suggestion
```

**Comment by apopiak** [comment]
Should I mirror it in the testing-runtime?

**Comment by jak-pan** [comment]
> Should I mirror it in the testing-runtime?

No

**Comment by apopiak** [comment]
> > Should I mirror it in the testing-runtime?
> 
> No. It's very short lived and we should be able to run stuff in testing runtime.

:ok: 
Damnit, thought you would say yes and abstracted it out :laughing:

**Comment by apopiak** [comment]
Seems like the re-run got stuck? Let's try another one?

**Comment by jak-pan** [comment]
I think since we changed the launch strategy to launch asap with basilisk marketplace and get into xcm UI we can no longer merge this... Is it the case? @lumir-mrkva

**Pull request closed without merging**